src/sentry/static/sentry/app/components/events/meta/metaProxy.jsx

```diff
@@ -29,34 +30,50 @@ export class MetaProxy {
       };
     }
 
+    // this is how  we can determine if current `obj` is a proxy
+    if (prop === IS_PROXY) {
+      return true;
+    }
+
+    const value = Reflect.get(obj, prop, receiver);
     if (
-      !obj.hasOwnProperty(prop) ||
-      typeof obj[prop] !== 'object' ||
-      _.isNull(obj[prop])
+      !Reflect.has(obj, prop, receiver) ||
+      typeof value !== 'object' ||
+      _.isNull(value)
     ) {
-      return obj[prop];
+      return value;
+    }
+
+    // This is so we don't create a new Proxy from an object that is
+    // already a proxy. Otherwise we can get into very deep recursive calls
+    if (Reflect.get(obj, IS_PROXY, receiver)) {
+      return value;
     }
 
     // Make sure we apply proxy to all children (objects and arrays)
     // Do we need to check for annotated inside of objects?
-    return new Proxy(
-      obj[prop],
-      new MetaProxy(this._meta, this.local && this.local[prop])
-    );
+    return new Proxy(value, new MetaProxy(this._meta, this.local && this.local[prop]));
   }
 }
 
 export function withMeta(event) {
-  if (!event) return null;
+  if (!event) {
+    return null;
+  }
 
   // Return unproxied `event` if browser does not support `Proxy`
-  if (typeof window.Proxy === 'undefined') return event;
+  if (typeof window.Proxy === 'undefined' || typeof window.Reflect === 'undefined') {
```

Comment:
Are we going to polyfill this somehow? We might make changes soon that might require rendering meta data in certain places to show errors to the user.

Reply:
Let's polyfill when we're ready to make those changes, we'll have to refactor Reflect out, but should be simple.